objects/src/notes/consumable.rs

```diff
@@ -0,0 +1,134 @@
+use miden_crypto::{
+    utils::{ByteReader, ByteWriter, Deserializable, Serializable},
+    Word,
+};
+use vm_processor::DeserializationError;
+
+use super::{NoteId, NoteMetadata, NoteRecipient, NoteVault, Nullifier};
+use crate::{Digest, Hasher};
+
+// CONSUMABLE NOTE
+// ================================================================================================
+
+/// A note with all the data required for it to be consumed by executing it against the transaction
+/// kernel.
+///
+/// Notes are created with a script, inputs, assets, and a serial number. Fungible and non-fungible
+/// asset transfers are done by moving assets to the note's vault. The note's script determines the
+/// conditions required for the note consumpution, i.e. the target account of a P2ID or conditions
+/// of a SWAP, and the effects of the note. The serial number has a double duty of preventing double
+/// spend, and providing unlikability to the consumer of a note. The note's inputs allow for
+/// customization of its script.
+///
+/// To create a note, the kernel does not require all the information above, a user can create a
+/// note only with the commitment to the script, inputs, the serial number, and the kernel only
+/// verifies the source account has the assets necessary for the note creation. See [NoteRecipient]
+/// for more details.
+#[derive(Clone, Debug, PartialEq, Eq)]
+#[cfg_attr(feature = "serde", derive(serde::Deserialize, serde::Serialize))]
+pub struct NoteConsumable {
+    vault: NoteVault,
+    metadata: NoteMetadata,
+    recipient: NoteRecipient,
+
+    id: NoteId,
+    nullifier: Nullifier,
+}
```

Comment:
IMO we should stop putting stuff in the `mod.rs`, specially when we have sub modules. There is no need to put everything in the `mod.rs`, and having sub mods makes it easier to navigate the source code.

Reply:
renamed and moved to `mod.rs`

Reply:
> IMO we should stop putting stuff in the `mod.rs`, specially when we have sub modules. There is no need to put everything in the `mod.rs`, and having sub mods makes it easier to navigate the source code.

I'm not opposed to this. Sometimes, when the inner and outer module names are the same, clippy may give an "inception" warning - but I don't think that's the case here (because `notes` and `note` would be different).